src/planet-a/Active/SysPlanetA.cpp

```diff
@@ -210,9 +245,123 @@ void SysPlanetA::update_geometry()
                                  MeshVisualizer3D::Normal{})
                 .setIndexBuffer(planet.m_indxBufGL, 0,
                                 MeshIndexType::UnsignedInt)
-                .setCount(planet.m_planet.calc_index_count());
+                .setCount(rPlanetGeo.calc_index_count());
+        }
+
+        if (m_debugUpdate.triggered() || true)
+        {
+            planet_update_geometry(ent);
+        }
+    }
+}
+
+void SysPlanetA::planet_update_geometry(osp::active::ActiveEnt planetEnt)
+{
+    auto &rPlanetPlanet = m_scene.reg_get<ACompPlanet>(planetEnt);
+    auto const &planetTf = m_scene.reg_get<ACompTransform>(planetEnt);
+    auto const &planetActivated = m_scene.reg_get<ACompActivatedSat>(planetEnt);
+
+    // TODO: de-spaghettify systems. make systems entirely stateless
+    osp::universe::Universe const &uni = m_scene.dynamic_system_find<SysAreaAssociate>().get_universe();
+
+    Satellite planetSat = planetActivated.m_sat;
+    auto &planetUComp = uni.get_reg().get<universe::UCompPlanet>(planetSat);
+
+    PlanetGeometryA &rPlanetGeo = *(rPlanetPlanet.m_planet);
+
+    // Temporary: use the first camera in the scene as the viewer
+    ActiveEnt cam = m_scene.get_registry().view<ACompCamera>().front();
+    auto const &camTf = m_scene.reg_get<ACompTransform>(cam);
+
+    // Set this somewhere else. Radius at which detail falloff will start
+    // This will essentially be the physics area size
+    float viewerRadius = 64.0f;
```

Comment:
looks like a good candidate for global.h for the initial value.

Reply:
It seems like it's specific to the planet rendering stuff and may be changed via graphics settings so I think keeping it in this file is appropriate, or perhaps moving it to the header

Reply:
Right now, the most practical place to put this is in SysAreaAssociate, as it describes the physics area. However, I'm soon planning on removing states from the IDynamicSystem classes, possibly creating a component struct to store it instead. I'd rather keep this variable as-is, as too many other things around it will change.